core/src/main/scala/zio/jdbc/ZConnectionPool.scala

```diff
@@ -165,14 +165,16 @@ object ZConnectionPool {
         pool   <- ZPool.make(getConn, Range(config.minConnections, config.maxConnections), config.timeToLive)
         tx      = ZLayer.scoped {
                     for {
-                      connection <- pool.get
-                      _          <- ZIO.addFinalizerExit {
-                                      case Exit.Success(_) => ZIO.unit
-                                      case Exit.Failure(_) => connection.rollback.ignoreLogged
-                                    }
+                      connection     <- pool.get
+                      autoCommitMode <- ZIO.attempt(connection.connection.getAutoCommit)
+                      _              <- ZIO.addFinalizerExit {
+                                          case Exit.Success(_) =>
+                                            ZIO.unit
+                                          case Exit.Failure(_) =>
+                                            ZIO.unless(autoCommitMode)(connection.rollback.ignoreLogged)
```

Comment:
I need help understanding how to set up a test case to reach these lines. I tried different stuff, but all of them are making the entire transaction fail completely without getting to this point.

Reply:
Even if the transaction fails, the finalizer should be run when the transaction closes. How do you know it doesn't make it to this point?

Reply:
Thanks, John! I've sorted this out. It was my mistake.